x-pack/plugin/enrich/src/test/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunnerTests.java

```diff
@@ -303,6 +652,6 @@ public void onFailure(Exception e) {
         ShardSegments shard = shardSegments.getShards()[0];
         assertThat(shard.getSegments().size(), is(equalTo(1)));
         Segment segment = shard.getSegments().iterator().next();
-        assertThat(segment.getNumDocs(), is(equalTo(3)));
+        assertThat(segment.getNumDocs(), is(equalTo(expectedDocs)));
```

Comment:
maybe also add a thing test that configures object fields with a type (`type: object`)?